packages/forms-web-app/src/controllers/eligibility/planning-department.js

```diff
@@ -1,31 +1,67 @@
+const logger = require('../../lib/logger');
+const { getDepartmentFromName } = require('../../services/department.service');
+const { getDepartmentFromId } = require('../../services/department.service');
+const { createOrUpdateAppeal } = require('../../lib/appeals-api-wrapper');
 const { getDepartmentData } = require('../../services/department.service');
 const { VIEW } = require('../../lib/views');
 
 exports.getPlanningDepartment = async (req, res) => {
   const { departments } = await getDepartmentData();
-
+  const { appeal } = req.session;
+  let appealLPD = '';
+  if (appeal.lpaCode) {
+    const lpd = await getDepartmentFromId(appeal.lpaCode);
+    if (lpd) {
+      appealLPD = lpd.name;
+    }
+  }
   res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT, {
+    appealLPD,
     departments,
   });
 };
 
 exports.postPlanningDepartment = async (req, res) => {
   const { body } = req;
   const { errors = {}, errorSummary = [] } = body;
+  const { appeal } = req.session;
+  const { departments, eligibleDepartments } = await getDepartmentData();
 
   if (errors['local-planning-department']) {
     const errorMessage = errors['local-planning-department'].msg;
-    const { departments, eligibleDepartments } = await getDepartmentData();
 
-    if (errorMessage === 'Ineligible Department') {
-      res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT_OUT, eligibleDepartments);
-    } else {
-      res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT, { departments, errors, errorSummary });
+    if (errorMessage !== 'Ineligible Department') {
+      res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT, {
+        appealLPD: '',
+        departments,
+        errors,
+        errorSummary,
+      });
+      return;
     }
+    res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT_OUT, eligibleDepartments);
+  }
+
+  const lpaName = body['local-planning-department'];
+  const lpa = await getDepartmentFromName(lpaName);
+
+  appeal.lpaCode = lpa.id;
+  try {
+    req.session.appeal = await createOrUpdateAppeal(appeal);
+  } catch (e) {
+    logger.error(e);
+
+    res.render(VIEW.ELIGIBILITY.PLANNING_DEPARTMENT, {
+      appeal,
+      errors,
+      errorSummary: [{ text: e.toString(), href: '#' }],
+    });
     return;
   }
 
-  res.redirect(`/${VIEW.ELIGIBILITY.LISTED_BUILDING}`);
+  if (!errors['local-planning-department']) {
```

Comment:
We have the the check for ` if (errors['local-planning-department'])` on line 33.

If we get to line 62, is it possible we might not trigger this conditional? 

If we don't trigger the conditional, the controller wouldn't return a response (?) so we'd be in a bad state. Can we remove this conditional check, or add in another final fall back?

Reply:
The final one is because we could be in this section if the department is ineligible as we want still want to store the lpaCode in the appeal but at this point the render is already triggered. All the errors are caught except this one.